gwt-i18n/src/main/java/org/gwtproject/i18n/shared/Localizable.java

```diff
@@ -0,0 +1,119 @@
+ * If a type <code>Type</code> directly extends or implements
+ * <code>Localizable</code> (as opposed to
+ * {@link org.gwtproject.i18n.client.Constants} or
+ * {@link org.gwtproject.i18n.client.Messages}) and the following code is used
+ * to create an object from <code>Type</code> as follows:
+ *
+ * <pre class="code">Type localized = (Type)GWT.create(Type.class);</pre>
+ *
+ * then <code>localized</code> will be assigned an instance of a localized
+ * subclass, selected based on the value of the <code>locale</code> client
+ * property. The choice of subclass is determined by the following naming
+ * pattern:
+ *
+ * <table>
+ *
+ * <tr>
+ * <th align='left'>If <code>locale</code> is...&#160;&#160;&#160;&#160;</th>
+ * <th align='left'>The substitute class for <code>Type</code> is...</th>
+ * </tr>
+ *
+ * <tr>
+ * <td><i>unspecified</i></td>
+ * <td><code>Type</code> itself, or <code>Type_</code> if <code>Type</code>
+ * is an interface</td>
+ * </tr>
+ *
+ * <tr>
+ * <td><code>x</code></td>
+ * <td>Class <code>Type_x</code> if it exists, otherwise treated as if
+ * <code>locale</code> were <i>unspecified</i></td>
+ * </tr>
+ *
+ * <tr>
+ * <td><code>x_Y</code></td>
+ * <td>Class <code>Type_x_Y</code> if it exists, otherwise treated as if
+ * <code>locale</code> were <code>x</code></td>
+ * </tr>
+ *
+ * </table>
+ *
+ * where in the table above <code>x</code> is a <a
+ * href="http://ftp.ics.uci.edu/pub/ietf/http/related/iso639.txt">ISO language
+ * code</a> and <code>Y</code> is a two-letter <a
+ * href="http://userpage.chemie.fu-berlin.de/diverse/doc/ISO_3166.html">ISO
+ * country code</a>.
+ *
+ * <h3>Specifying Locale</h3>
+ * The locale of a module is specified using the <code>locale</code> client
+ * property, which can be specified using either a meta tag or as part of the
+ * query string in the host page's URL. If both are specified, the query string
+ * takes precedence.
+ *
+ * <p>
+ * To specify the <code>locale</code> client property using a meta tag in the
+ * host HTML, use <code>gwt:property</code> as follows:
+ *
+ * <pre>&lt;meta name="gwt:property" content="locale.new=x_Y"&gt;</pre>
+ *
+ * For example, the following host HTML page sets the locale to "ja_JP":
+ *
+ * {@gwt.include com/google/gwt/examples/i18n/ColorNameLookupExample_ja_JP.html}
+ * </p>
+ *
+ * <p>
+ * To specify the <code>locale</code> client property using a query string,
+ * specify a value for the name <code>locale</code>. For example,
+ *
+ * <pre>http://www.example.org/myapp.html?locale=fr_CA</pre>
+ *
+ * </p>
+ *
+ * <h3>For More Information</h3>
+ * See the GWT Developer Guide for an introduction to internationalization.
+ *
+ * @see org.gwtproject.i18n.client.Constants
+ * @see org.gwtproject.i18n.client.ConstantsWithLookup
+ * @see org.gwtproject.i18n.client.Messages
+ * @see org.gwtproject.i18n.client.Dictionary
+ */
+@Localizable.I18nLocaleSuffuxes({"default", "en", "es", "fr", "de"})//TODO real list
```

Comment:
since we are listing all possible locales here, shouldnt this interface be generated from CLDR data?!

Reply:
Good point, let me think on that some more. This certainly would suggest that we put the interface back in gwt-i18n-cldr, and make the cldr importer generate it - but we still shouldn't inherit it from those other classes.

If we do as you suggest, we can also remove the TODO in the processor about sorting the order, since we can just make the cldr pre-sort the order?

Reply:
About the order, yes true, since they will be ordered already.

Reply:
I'll add this to the gwt-i18n tracking doc